lib/usb_connection.js

```diff
@@ -189,11 +190,10 @@ USB.Connection.prototype.end = function() {
     // Tell the USB daemon to end all processes active
     // on account of this connection
     Daemon.deregister(self, function(err) {
-      self._close();
       if (err) {
         reject(err);
       } else {
-        resolve();
+        self._close(resolve());
```

Comment:
This won't work...

self._close(function (err) {
  if (err) reject(err)

  resolve()
})